x/group/types_test.go

```diff
@@ -0,0 +1,239 @@
+package group_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/x/group"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestPercentageDecisionPolicyAllow(t *testing.T) {
+	testCases := []struct {
+		name           string
+		policy         *group.PercentageDecisionPolicy
+		tally          *group.Tally
+		totalPower     string
+		votingDuration time.Duration
+		result         group.DecisionPolicyResult
+	}{
+		{
+			"YesCount percentage >= decision policy percentage",
+			&group.PercentageDecisionPolicy{
+				Percentage: "0.5",
+				Timeout:    time.Second * 100,
+			},
+			&group.Tally{
+				YesCount:     "2",
+				NoCount:      "0",
+				AbstainCount: "0",
+				VetoCount:    "0",
+			},
+			"3",
+			time.Duration(time.Second * 50),
+			group.DecisionPolicyResult{
+				Allow: true,
+				Final: true,
+			},
+		},
+		{
```

Comment:
```go
		{
			"No+Veto+Abstain percentage >= decision policy percentage",
			&group.PercentageDecisionPolicy{
				Percentage: "0.5",
				Timeout:    time.Second * 100,
			},
			&group.Tally{
				YesCount:     "2",
				NoCount:      "1",
				AbstainCount: "1",
				VetoCount:    "1",
			},
			"6",
			time.Duration(time.Second * 50),
			group.DecisionPolicyResult{
				Allow: false,
				Final: true,
			},
		},
```

The `Allow: false` result also depends on, I think we should go with 51% yes count.